packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart

```diff
@@ -379,6 +393,62 @@ class _StoryOfYourFlexWidgetState extends State<StoryOfYourFlexWidget> {
         ),
       ),
     );
+
+    // cross axis does not have any empty space
+    if (crossAxisDimension(childProperties) == crossAxisDimension(properties) ||
+        crossAxisAlignment == CrossAxisAlignment.stretch) {
+      return renderChildWidget;
+    }
+
+    // cross axis have empty space so we need to visualize it
+    double emptySpaceRenderWidth, emptySpaceRenderHeight;
+    double emptySpaceWidth, emptySpaceHeight;
+    if (direction == Axis.horizontal) {
+      emptySpaceWidth = childProperties.width;
+      emptySpaceRenderWidth = renderWidth;
+
+      emptySpaceHeight = properties.height - childProperties.height;
+
+      emptySpaceRenderHeight = max(maxHeight - renderHeight, 0);
+      if (crossAxisAlignment == CrossAxisAlignment.center) {
+        emptySpaceHeight /= 2.0;
+        emptySpaceRenderHeight /= 2.0;
+      }
+    } else {
+      emptySpaceHeight = childProperties.height;
+      emptySpaceRenderHeight = renderHeight;
+
+      emptySpaceWidth = properties.width - childProperties.width;
+      emptySpaceRenderWidth = max(maxWidth - renderWidth, 0.0);
+      if (crossAxisAlignment == CrossAxisAlignment.center) {
+        emptySpaceWidth /= 2.0;
```

Comment:
there is getting to be a lot of similar logic for width and height throughout. Consider adding some helper classes that track this data along a single axis and reusing them to reduce the amount of duplication. That is ok to do as a follow up cl after updating the goldens to make sure that they refactor does not break the functionality. we can sync up in person for some ideas on ways to increase the code reuse without making the code harder to read.